app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java

```diff
@@ -8,24 +8,29 @@
 import java.util.InputMismatchException;
 import java.util.Scanner;
 
+import static android.location.Location.distanceBetween;
+
+
 // Assumption is no two points are equivalent
 public class Graph
 {
     private int numberOfNodes;
     private Node[] nodes;
     private N_aryTree breathFirstSearchResults;
 
-    private class Node
+    public class Node
     {
         private int id;
         private LatLng element;
         private LinkedList<Node> adjacencyList;
         private boolean traversed;
+        private int type;  // 0 is a class node, 1 is a hall node.
```

Comment:
What is a hall node? Is it simply a node between corridors and hallways?

Reply:
Yes a class node is any node that is positioned on a class room.
A hall node is any node that is positioned in the hallways. 
In a search we look at the starting class node and ending class node, and get the path between the 2 using only hall nodes.